### PR TITLE
PRIME-2294 New API for Medinet

### PR DESCRIPTION
**Pull request by bergomi02**
- The new API works similar to existing 'gpid' endpoint but return the detail enrollee information as HpdidLookup API.

PR Endpoint: https://pr-2203.pharmanetenrolment.gov.bc.ca/api/v1/provisioner-access/gpid-detail

Note: 
- Please find the testing instruction in Teams:
  PRIME Developers -> Files -> Testing Tools & Instructions -> "PRIME - Testing GetGpid with Postman.docx"

### CODE REVIEW

File: prime-dotnet-webapi/Controllers/ProvisionerAccessController.cs [hunk 1]
```diff
@@ -136,6 +136,19 @@ public async Task<ActionResult> GetGpid()
...
         }
 
+        // GET: api/provisioner-access/gpid-detail
+        /// <summary>
+        /// Gets the GPID and detail info for the user. Only a valid token is required, no role is required.
+        /// </summary>
+        [HttpGet("gpid-detail", Name = nameof(GetGpidDetail))]
+        [Authorize]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(typeof(ApiResultResponse<string>), StatusCodes.Status200OK)]
```
Comment (neophyte57) [inline prime-dotnet-webapi/Controllers/ProvisionerAccessController.cs:146]: I think this is only meta-data for Swagger but should that be `[ProducesResponseType(typeof(ApiResultResponse<HpdidLookup>), StatusCodes.Status200OK)]`?

Comment (bergomi02) [reply prime-dotnet-webapi/Controllers/ProvisionerAccessController.cs:146]: Good catch. it is fixed.

File: prime-dotnet-webapi/Services/EnrolleeService.cs [hunk 2]
```diff
@@ -93,6 +93,39 @@ public async Task<string> GetActiveGpidAsync(Guid userId)
...
+        /// <param name="userId"></param>
+        public async Task<HpdidLookup> GetActiveGpidDetailAsync(Guid userId)
+        {
+            return await _context.Enrollees
+                .Where(enrollee => enrollee.UserId == userId
+                    && enrollee.CurrentStatus.StatusCode != (int)StatusType.Declined)
+                .Select(e => new HpdidLookup
+                {
+                    Gpid = e.GPID,
+                    Hpdid = e.HPDID,
```
Comment (neophyte57) [inline prime-dotnet-webapi/Services/EnrolleeService.cs:109]: Thinking about it more, for this API's client, we should **not** be returning HPDID

Comment (bergomi02) [reply prime-dotnet-webapi/Services/EnrolleeService.cs:109]: HPDID has been removed.

**Review by neophyte57 — APPROVED** [review]
🎉

Tested these scenarios
- Enrollee with signed TOA
- NP PRIME Enrollee with signed TOA
- Enrollee without TOA 
- Invalid bearer token 
- Declined Enrollee

**Pull request merged**